src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6474,6 +6474,11 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                                 Debug.Assert(!leftType.IsDynamic());
                                 Debug.Assert(IsPotentialColorColorReceiver(left, leftType));
 
+                                if (leftSymbol is SourceFieldSymbolWithSyntaxReference fieldLeft)
+                                {
+                                    ConstantFieldsInProgress.RemoveDepencency(fieldLeft);
```

Comment:
>ConstantFieldsInProgress.RemoveDepencency(fieldLeft); [](http://example.com/codeflow?start=36&length=53)

My original comment about this line is still relevant. the field should not be removed unconditionally. It can be removed only if ```BindIdentifier``` called above added it (i.e. it was not in the set before). Also, if later on we decide to resolve ```BoundTypeOrValueExpression``` to the ```boundValue```, the field should be put back into the set.

Reply:
I'm adding the first check you mentioned

> Also, if later on we decide to resolve BoundTypeOrValueExpression to the boundValue, the field should be put back into the set.

I don't get this part

Reply:
>>Also, if later on we decide to resolve BoundTypeOrValueExpression to the boundValue, the field should be put back into the set.

> I don't get this part

Eventually we figure out what the identifier should mean and replace ```BoundTypeOrValueExpression``` with ```boundValue``` or ```BoundTypeExperession```. Here we are preparing to resolve the Color Color, eventually we will resolve it. ```BoundTypeOrValueExpression``` is a transient node, it simply means: "I do not know yet what I represent, but I know my type". If we resolve it to a value (instead of a type) and that value is supposed to be in dependencies, we should update the dependencies accordingly.

Reply:
In other words, we are removing the ```fieldLeft``` from dependencies here not because it should not be treated as a dependency, but because we don't know yet whether it is a dependency. Eventually we will know if it is a dependency.

Reply:
Right, so you mean after resolving the Color Color case this might be added back, in which case do we want to remove the dependency preemptively, or keep track of it and remove it once? The latter sounds more complicated though I still think.

Reply:
Later it might be harder to determine if the dependency should be removed. Who knows who else is adding them. It is much safer to add a dependency later, even if it is already in the set.